core/src/backends/lnd/rest/types.rs

```diff
@@ -253,3 +255,65 @@ impl TryInto<PayInvoiceResult> for SendPaymentSyncResponse {
         Ok(result)
     }
 }
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DecodeInvoiceResponse {
+    pub destination: String,
+    pub payment_hash: String,
+    pub num_satoshis: String,
+    pub timestamp: String,
+    pub expiry: String,
+    pub description: String,
+    pub description_hash: String,
+    pub fallback_addr: String,
+    pub cltv_expiry: String,
+    pub route_hints: Vec<Value>,
+    pub payment_addr: String,
+    pub num_msat: String,
+    pub features: HashMap<u16, DecodeInvoiceFeature>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DecodeInvoiceFeature {
+    pub name: String,
+    pub is_required: bool,
+    pub is_known: bool,
+}
+
+fn extract_feature_status(feature: Option<&DecodeInvoiceFeature>) -> FeatureActivationStatus {
+    if feature.is_none() {
+        return FeatureActivationStatus::Unknown;
+    }
+
+    let _feature = feature.unwrap();
+
+    match (_feature.is_known, _feature.is_required) {
+        (true, true) => FeatureActivationStatus::Mandatory,
+        (true, false) => FeatureActivationStatus::Optional,
+        _ => FeatureActivationStatus::Unknown,
+    }
```

Comment:
Untested but I guess you could do something like this instead:
```suggestion
    match feature {
        None => FeatureActivationStatus::Unknown,
        Some(f) => match (f.is_known, f.is_required) {
            (true, true) => FeatureActivationStatus::Mandatory,
            (true, false) => FeatureActivationStatus::Optional,
            _ => FeatureActivationStatus::Unknown,
    }
```

Reply:
yes seems to be working properly